src/types/register.ts

```diff
@@ -21,4 +23,7 @@ export type RegisteredRejectionType = Register extends { rejections: infer TReje
 
 export type RegisteredRouterState = Register extends { state: infer TState }
   ? TState
-  : {}
\ No newline at end of file
+  : {}
+
+export type RegisteredRouteMap = RoutesMap
```

Comment:
I think we should just get rid of the RoutesMap type and define RegisteredRoutesMap here in this file directly.